tests/publiccloud/aistack_basic.pm

```diff
@@ -0,0 +1,186 @@
+    process_reboot(trigger => 1);
+
+    for (split(',\s*', $runtime)) {
+        my $package = $_;
+        my $pkg_status = script_run("rpm -q $package");
+        my $tr_install_cmd = "";
+        record_info("Package status $pkg_status");
+
+        if ($pkg_status == 0) {
+            record_info("$package is already installed");
+        } else {
+            record_info("$package is not installed, installing now...");
+            if ($package eq "rke2-server") {
+                trup_call("run bash -c 'curl -sSL $rke2_url | sudo sh'");
+                record_info("Installed $package.");
+                process_reboot(trigger => 1);
+            } elsif ($package eq "helm") {
+                trup_call("run bash -c 'curl -sSL $helm_url | sudo sh'");
+                record_info("Installed $package.");
+                process_reboot(trigger => 1);
+                # $tr_install_cmd .= "curl -sSL $rke2_url | sudo sh; ";
+                # $tr_install_cmd .= "curl -sSL $helm_url | sudo sh; ";
+            } elsif ($package eq "kubectl") {
+                trup_call("run bash -c 'curl -sSL $kubectl_url | sudo sh'");
+                record_info("Installed $package.");
+                process_reboot(trigger => 1);
+                # $tr_install_cmd .= "curl -LO $kubectl_url | sudo sh; ";
+            }
+            if (($package eq "docker") || ($package eq "curl") || ($package eq "git")) {
+                trup_call("pkg install $package");
+                record_info("Installed $package.");
+                process_reboot(trigger => 1);
+            }
+        }    # pkg_status if
+             #if (($package eq "rke2-server") || ($package eq "kubectl") || ($package eq "helm")) {
+             #    my $script_file = "$package.sh";
+             #    tr_run_script($script_file, $tr_install_cmd);
+
+        # script_run("echo '#!/bin/bash' > ~/$script_file");
+        # script_run("echo 'set -e' >> ~/$script_file");
+        # script_run("echo '$tr_install_cmd' >> ~/$script_file");
+        # script_run("chmod 755 ~/$script_file");
+        # script_run("cat ~/$script_file");
+        ## Execute the script using trup_run
+        # trup_call("run bash -c ~/$script_file");
+        # record_info("Executed script for $package.");
+        # Delete the script
+        # script_run("rm ~/$script_file");
+        # record_info("Executed script for $package.");
+        #}
+        # process_reboot(trigger => 1);
+    }    # for loop
+
+    for (split(/,\s*/, $runtime)) {
+        my $ins_package = $_;
+        if ($ins_package eq 'rke2-server' || $ins_package eq 'docker') {
+            record_info("$ins_package is installed. Enable and start server");
+            script_run("sudo systemctl enable $ins_package", timeout => 100);
+            script_run("sudo systemctl start $ins_package", timeout => 100);
+            script_run("sudo systemctl status $ins_package", timeout => 100);
+        }
+    }
+}
+
+sub install_aistack_chart {
```

Comment:
Also, in addition to what `foursixnine` pointed out, pay attention to eventual `_SECRET` steps, if any: we tend to avoid hardcoding _sensible_ `keys` :slightly_smiling_face:

Reply:
fixed